hibernate-core/src/main/java/org/hibernate/sql/ast/spi/AbstractSqlAstTranslator.java

```diff
@@ -1196,6 +1196,15 @@ else if ( assignedValue instanceof SelectStatement ) {
 		}
 	}
 
+	private void appendColumn(ColumnReference column) {
```

Comment:
Might be better to let subclasses override this when they support qualifying columns in the set clause and default to `appendColumnForWrite( this, null )` i.e. drop the qualifier.